APIv4 - Deprecate unnecessary event

colemanw opened the pull request:
Overview
----------------------------------------
Makes it easier for extensions to hook into APIv4 entities by simplifying the events called.

Before
----------------------------------------
To implement a virtual API4 entity, an extension must listen to 2 events, one of which is redundant.

After
----------------------------------------
Redundant event no longer used, as the one event returns all necessary data.

Technical Details
----------------------------------------
All the info from `civi.api4.entityTypes` is sufficient to create an APIv4 request.
`civi.api4.createRequest` was redundant.

colemanw committed:
APIv4 - Deprecate unnecessary event

Before: To implement a virtual API4 entity, an extension must listen to 2 events, one of which is redundant.
After: Redundant event no longer used, as the one event returns all necessary data.

All the info from `civi.api4.entityTypes` is sufficient to create an APIv4 request.
`civi.api4.createRequest` was redundant.

colemanw commented:
@jensschuppe I confirmed that the unit tests in ECK are still passing after this change.

seamuslee001 commented:
This seems fine to me

eileenmcnaughton commented:
unit tests might be on board but the style police aren't.... https://test.civicrm.org/job/CiviCRM-Core-PR/48340/checkstyle/

colemanw committed:
APIv4 - Move filtering of component entities higher up

Filtering out component entities at a low level was causing errors when trying
to get basic info about an entity from a disabled component.
It's better to do the filtering in the Entity.get action instead.
